### ISSUE TITLE
TypeError "<" not supported between instances of type "str" and "NoneType"

### ISSUE THREAD

**Issue by bebetoalves** [archlinux/archinstall#813]
I tried to do manual partitioning, as automatic partitioning doesn't offer an option to not create the "home" partition. And it was like this:

![20211227_094417](https://user-images.githubusercontent.com/13764984/147473693-c773e06a-ca38-4add-addf-f4191e85483e.jpg)
![20211227_094515](https://user-images.githubusercontent.com/13764984/147473694-57559e89-f85f-40b2-87c6-82f7d0e085b2.jpg)

However, when formatting the partitions this error occurred:

![20211227_094638](https://user-images.githubusercontent.com/13764984/147473762-4b2d87b1-38c9-4546-8c0d-fd56ac1bbafd.jpg)

**Comment by Torxed** [comment]
The main reason being `/boot` doesn't have a `mountpoint` set.
I'm not sure why, so I would need to know which steps you took to end up here.

The error itself, was fixed in #783 with the same issue title. But again, your particular issue is a bit different but I won't know why it happened without some more information.

**Comment by bebetoalves** [comment]
> The main reason being `/boot` doesn't have a `mountpoint` set. I'm not sure why, so I would need to know which steps you took to end up here.
> 
> The error itself, was fixed in #783 with the same issue title. But again, your particular issue is a bit different but I won't know why it happened without some more information.

The only different thing I did was not set the 500MB partition to the /boot mount point because I ran into an error, already reported in #812 . I did the same process using the stable version of archinstall and it worked.

**Comment by wllacer** [comment]
Did anyone tested it with the new 2.3.1.dev  or master?  
The reason behind is that somehow unmountable partitions slipped in the list of partitions to be mounted, and a sort routine freaked out. @Torxed has corrected it in the above mentioned versions (just a couple of days ago)

**Comment by Torxed** [comment]
> Did anyone tested it with the new 2.3.1.dev or master? The reason behind is that somehow unmountable partitions slipped in the list of partitions to be mounted, and a sort routine freaked out. @Torxed has corrected it in the above mentioned versions (just a couple of days ago)

I think it's fixed. The thing that spook me a bit and why I haven't closed this issue.
Is because the installer continued without a `/boot` which would break the installation further down the line. I'm still wondering how that could happen.

**Comment by wllacer** [comment]
Unless It is a btrfs partition, once the process enters
mount_ordered_layout partition['mountpoint'] will not be touched (Believe
me, i've dealt almost only with the routine the last weeks).
Unless there is a freak side-effect, It has to come this way already wrong.
Moreso, the disk layout file is written to disk before the installer
instantiation. So the most probable cause for the boot problem comes during
the data adquisition process.
An install.log would be welcome

**Comment by getong** [comment]
```
{
    "audio": null,
    "bootloader": "grub-install",
    "desktop-environment": "gnome",
    "dry-run": false,
    "gfx_driver": "All open-source (default)",
    "harddrives": [
        "/dev/sda"
    ],
    "hostname": "archlinux",
    "kernels": [
        "linux"
    ],
    "keyboard-layout": "es",
    "mirror-region": {
        "China": {
            "http://mirror.lzu.edu.cn/archlinux/$repo/os/$arch": true,
            "http://mirror.redrock.team/archlinux/$repo/os/$arch": true,
            "http://mirrors.163.com/archlinux/$repo/os/$arch": true,
            "http://mirrors.aliyun.com/archlinux/$repo/os/$arch": true,
            "http://mirrors.bfsu.edu.cn/archlinux/$repo/os/$arch": true,
            "http://mirrors.cqu.edu.cn/archlinux/$repo/os/$arch": true,
            "http://mirrors.dgut.edu.cn/archlinux/$repo/os/$arch": true,
            "http://mirrors.hit.edu.cn/archlinux/$repo/os/$arch": true,
            "http://mirrors.neusoft.edu.cn/archlinux/$repo/os/$arch": true,
            "http://mirrors.nju.edu.cn/archlinux/$repo/os/$arch": true,
            "http://mirrors.tuna.tsinghua.edu.cn/archlinux/$repo/os/$arch": true,
            "http://mirrors.ustc.edu.cn/archlinux/$repo/os/$arch": true,
            "http://mirrors.zju.edu.cn/archlinux/$repo/os/$arch": true,
            "https://mirror.redrock.team/archlinux/$repo/os/$arch": true,
            "https://mirrors.aliyun.com/archlinux/$repo/os/$arch": true,
            "https://mirrors.bfsu.edu.cn/archlinux/$repo/os/$arch": true,
            "https://mirrors.cqu.edu.cn/archlinux/$repo/os/$arch": true,
            "https://mirrors.dgut.edu.cn/archlinux/$repo/os/$arch": true,
            "https://mirrors.hit.edu.cn/archlinux/$repo/os/$arch": true,
            "https://mirrors.neusoft.edu.cn/archlinux/$repo/os/$arch": true,
            "https://mirrors.nju.edu.cn/archlinux/$repo/os/$arch": true,
            "https://mirrors.sjtug.sjtu.edu.cn/archlinux/$repo/os/$arch": true,
            "https://mirrors.tuna.tsinghua.edu.cn/archlinux/$repo/os/$arch": true,
            "https://mirrors.ustc.edu.cn/archlinux/$repo/os/$arch": true,
            "https://mirrors.xjtu.edu.cn/archlinux/$repo/os/$arch": true
        }
    },
    "nic": {
        "nic": "enp2s0"
    },
    "ntp": true,
    "packages": [
        "vim",
        "v2ray"
    ],
    "profile": {
        "path": "/usr/lib/python3.10/site-packages/archinstall/profiles/desktop.py"
    },
    "script": "guided",
    "swap": true,
    "sys-encoding": "utf-8",
    "sys-language": "en_US",
    "timezone": "Asia/Shanghai"
}
{
    "/dev/sda": {
        "partitions": [
            {
                "ESP": false,
                "PARTUUID": "c66f100e-86f3-4300-a61c-50a15ab7b49c",
                "boot": false,
                "encrypted": false,
                "filesystem": {
                    "format": ""
                },
                "mountpoint": null,
                "size": 2048,
                "start": 2048,
                "type": "primary",
                "wipe": false
            },
            {
                "ESP": false,
                "PARTUUID": "e0bfe48c-f7b5-8645-bac6-25e7a84353f0",
                "boot": false,
                "encrypted": false,
                "filesystem": {
                    "format": "swap"
                },
                "mountpoint": null,
                "size": 268435456,
                "start": 4096,
                "type": "primary",
                "wipe": false
            },
            {
                "ESP": false,
                "PARTUUID": "46089485-484a-5e44-9cb8-18c86d4d8840",
                "boot": false,
                "encrypted": false,
                "filesystem": {
                    "format": "btrfs"
                },
                "format": true,
                "mountpoint": "/",
                "size": 2147483648,
                "start": 268439552,
                "type": "primary",
                "wipe": false
            },
            {
                "ESP": false,
                "PARTUUID": "5dcec055-8e25-e847-872a-19b4be345252",
                "boot": false,
                "encrypted": false,
                "filesystem": {
                    "format": "btrfs"
                },
                "mountpoint": null,
                "size": 5398111822,
                "start": 2415923200,
                "type": "primary",
                "wipe": false
            }
        ]
    }
}
Formatting /dev/sda3 -> btrfs
'<' not supported between instances of 'str' and 'NoneType'
Hardware model detected: Unknow Unknow; UEFI mode: False
```
I test the current master, c6fdf775c8b601a9b94f7c12849276e0b6fe0bfe , the problem is still not fixed.

**Comment by Torxed** [comment]
You don't happen to have the stack trace?
I must have missed another `sort()` somewhere and it would help narrow it down.
If not, I can most likely reproduce it with the conf you've supplied so it's fine if you don't have it :)

**Comment by Torxed** [comment]
I believe this might have been fixed in v2.3.2. If it remains, I'm willing to re-open, but I would probably need a stack trace :)
